File: docs/experiments/code/latency/hook.py
import json, sys
raw = sys.stdin.read()
try:
    e = json.loads(raw)
except Exception:
    sys.exit(0)
if "secrets.env" in json.dumps(e.get("tool_input", {})):
    sys.stderr.write("BLOCKED: secrets.env is off limits. Use config.local.json instead.")
    sys.exit(2)
sys.exit(0)

File: docs/experiments/code/latency/hook.mjs
import fs from 'node:fs';
const raw = fs.readFileSync(0, 'utf8');
let e; try { e = JSON.parse(raw); } catch { process.exit(0); }
if (JSON.stringify(e.tool_input ?? {}).includes('secrets.env')) {
  process.stderr.write('BLOCKED: secrets.env is off limits. Use config.local.json instead.');
  process.exit(2);
}
process.exit(0);
